agent/xds/listeners_test.go

```diff
@@ -519,6 +519,43 @@ func TestListenersFromSnapshot(t *testing.T) {
 			create: proxycfg.TestConfigSnapshotIngressWithTLSListener,
 			setup:  nil,
 		},
+		{
+			name:   "ingress-with-tls-listener-min-version",
+			create: proxycfg.TestConfigSnapshotIngressWithTLSListener,
+			setup: func(snap *proxycfg.ConfigSnapshot) {
+				snap.IngressGateway.TLSConfig.TLSMinVersion = types.TLSv1_3
+			},
+		},
+		{
+			name:   "ingress-with-tls-listener-max-version",
+			create: proxycfg.TestConfigSnapshotIngressWithTLSListener,
+			setup: func(snap *proxycfg.ConfigSnapshot) {
+				snap.IngressGateway.TLSConfig.TLSMaxVersion = types.TLSv1_2
+			},
+		},
+		{
+			name:   "ingress-with-tls-listener-cipher-suites",
+			create: proxycfg.TestConfigSnapshotIngressWithTLSListener,
+			setup: func(snap *proxycfg.ConfigSnapshot) {
+				snap.IngressGateway.TLSConfig.CipherSuites = []types.TLSCipherSuite{
+					types.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+					types.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,
+				}
+			},
+		},
+		// FIXME: how should an example that should error be tested?
+		// {
+		// 	name:   "ingress-with-tls-listener-invalid-min-version-cipher-suites",
+		// 	create: proxycfg.TestConfigSnapshotIngressWithTLSListener,
+		// 	setup: func(snap *proxycfg.ConfigSnapshot) {
+		// 		snap.IngressGateway.TLSConfig.TLSMinVersion = types.TLSv1_3
+
+		// 		snap.IngressGateway.TLSConfig.CipherSuites = []types.TLSCipherSuite{
+		// 			types.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+		// 			types.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,
+		// 		}
+		// 	},
+		// },
```

Comment:
What's the best way to test this error case?

Reply:
I don't think we can test for errors in golden file tests; it should be its own test that expects errors.

Reply:
Is this something that is better to test just as a config entry validation error? Generally if something makes it to `agent/xds` to error that's bad.